File: my-class-mate-be/src/main/java/com/bill/model/response/ParticipationResponse.java
package com.bill.model.response;

import com.bill.constant.ParticipationStatusEnum;
import lombok.*;
import lombok.experimental.FieldDefaults;

import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
public class ParticipationResponse {
    Long participationId;
    Long courseScheduleId;
    Integer round;
    String topic;
    ParticipationStatusEnum status;
    Long createdBy;
    LocalDateTime createdAt;
    LocalDateTime closedAt;
}

File: my-class-mate-fe/src/api/data/participation-response.ts
export interface ParticipationResponse {
    participationId: number;
    courseScheduleId: number;
    round: number;
    topic: string;
    status: ParticipationStatus;
    createdBy: number;
    createdAt: string;
    closedAt: string | null;
}

export type ParticipationStatus = 'OPEN' | 'CLOSED' | 'CLOSE';

export interface ParticipationListResponse {
    success: boolean;
    data: ParticipationResponse[];
    total: number;
    message: string;
}

export interface CreateParticipationRequest {
    courseScheduleId: number;
    lecturerId: number;
    topic: string;
}

export interface CreateParticipationResponse {
    success: boolean;
    data: ParticipationResponse;
    message: string;
}